Orange/widgets/widget.py

```diff
@@ -570,6 +570,39 @@ def progressBarFinished(self, processEvents=QEventLoop.AllEvents):
         if processEvents is not None and processEvents is not False:
             qApp.processEvents(processEvents)
 
+    @contextlib.contextmanager
+    def progressBar(self, iterations=0):
+        """
+        Context manager for progress bar.
+
+        Using it ensures that the progress bar is removed at the end without
+        needing the `finally` blocks.
+
+        Usage:
+
+            with self.progressBar(20) as progress:
+                ...
+                progress.advance()
+
+        or
+
+            with self.progressBar() as progress:
+                ...
+                progress.advance(15)
```

Comment:
Calling `advance()` without initializing with total number of iterations results in what? 1500%?